### PR TITLE
Warn when not enough FDs

### PR DESCRIPTION
**Pull request by erictune**
Helps debug problem when running tests on OS X.

### CODE REVIEW

**Comment by spxtr** [comment]
@k8s-bot unit test this again issue: #21188

File: hack/test-go.sh [hunk 1]
```diff
@@ -303,6 +303,14 @@ reportCoverageToCoveralls() {
   fi
 }
 
+checkFDs() {
+  # several unittests panic when httptest cannot open more sockets
+  # due to the low default files limit on OS X.  Warn about low limit.
+  fileslimit = $(ulimit -n)
+  if [[ fileslimit -lt 1000 ]]; then
+echo "WARNING: ulimit -n (files) should be at least 1000, is $fileslimit, may cause test failure"; fi
```
Comment (spxtr) [inline hack/test-go.sh]: Indentation is strange here. Should be
```bash
  if [[ fileslimit -lt 1000 ]]; then
     echo "WARNING: ulimit -n (files) should be at least 1000, is $fileslimit, may cause test failure"
  fi
```
I'm also not a huge fan of the warning message.

File: hack/test-go.sh [hunk 2]
```diff
@@ -303,6 +303,14 @@ reportCoverageToCoveralls() {
   fi
 }
 
+checkFDs() {
+  # several unittests panic when httptest cannot open more sockets
+  # due to the low default files limit on OS X.  Warn about low limit.
+  fileslimit = $(ulimit -n)
```
Comment (spxtr) [inline hack/test-go.sh]: This can be local.

File: hack/test-go.sh [hunk 3]
```diff
@@ -303,6 +303,14 @@ reportCoverageToCoveralls() {
   fi
 }
 
+checkFDs() {
```
Comment (ixdy) [inline hack/test-go.sh]: were you intending to call this from somewhere?

File: hack/test-go.sh (hunk 1 above)
Comment (erictune) [inline hack/test-go.sh]: Please suggest alternate warning message.  Or do you want it to just exit?

File: hack/test-go.sh (hunk 2 above)
Comment (erictune) [inline hack/test-go.sh]: ok

File: hack/test-go.sh (hunk 3 above)
Comment (erictune) [inline hack/test-go.sh]: fixed, thx.

**Comment by spxtr** [comment]
@k8s-bot unit test this issue #21257

**Comment by spxtr** [comment]
I'm worried that this will get lost in a sea of output logs unless it actually fails the test run. We currently print out way too much stuff, much of which already starts with WARNING or ERROR or FAIL.

**Comment by spxtr** [comment]
LGTM

**Pull request merged**